core/src/main/java/uk/gov/gchq/koryphe/impl/function/StringAppend.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.koryphe.impl.function;
+
+import org.apache.commons.lang3.StringUtils;
+
+import uk.gov.gchq.koryphe.Since;
+import uk.gov.gchq.koryphe.Summary;
+import uk.gov.gchq.koryphe.function.KorypheFunction;
+
+/**
+ * A {@code StringAppend} is a {@link java.util.function.Function} which takes a input {@link String} and returns the
+ * same string, appended with a suffix.
+ */
+@Since("1.8.3")
+@Summary("Appends a provided suffix to a string.")
+public class StringAppend extends KorypheFunction<String, String> {
+
+    private String suffix;
+
+    public StringAppend() {
```

Comment:
Because the validation is done in the setter, this function doesn't work if the suffix is never set. Instead of just returning the original string, it appends "null" to the end. If you move the logic into the apply function, this test will pass:
```java
    @Test
    public void shouldHandleUnsetSuffix() {
        // Given
        final StringAppend function = new StringAppend();

        // When
        final String result = function.apply("Hello");

        // Then
        assertEquals("Hello", result); // right now result is Hellonull
    }
``` 
The StringPrepend function also has this issue